ledger/store/trackerdb/interface.go

```diff
@@ -26,21 +26,39 @@ import (
 	"github.com/algorand/go-algorand/ledger/ledgercore"
 )
 
+// AccountRef is an opaque ref to an account in the db.
+type AccountRef interface {
+	AccountRefMarker()
+}
+
+// OnlineAccountRef is an opaque ref to an "online" account in the db.
+type OnlineAccountRef interface {
+	OnlineAccountRefMarker()
+}
+
+// ResourceRef is an opaque ref to a resource in the db.
+type ResourceRef interface {
+	ResourceRefMarker()
+}
```

Comment:
cool idea! this will certainly help keep us from using incompatible ref types.

Tagging @cce specifically because this could be a valuable pattern in other places and I'd like to highlight it. Specifically, our Peer interface is also an opaque ref